pythonforandroid/bootstrap.py

```diff
@@ -138,36 +139,52 @@ def run_distribute(self):
         self.distribution.save_info(self.dist_dir)
 
     @classmethod
-    def list_bootstraps(cls):
+    def all_bootstraps(cls):
         '''Find all the available bootstraps and return them.'''
         forbidden_dirs = ('__pycache__', 'common')
         bootstraps_dir = join(dirname(__file__), 'bootstraps')
+        result = set()
         for name in listdir(bootstraps_dir):
             if name in forbidden_dirs:
                 continue
             filen = join(bootstraps_dir, name)
             if isdir(filen):
-                yield name
+                result.add(name)
+        return result
 
     @classmethod
     def get_bootstrap_from_recipes(cls, recipes, ctx):
         '''Returns a bootstrap whose recipe requirements do not conflict with
         the given recipes.'''
         info('Trying to find a bootstrap that matches the given recipes.')
         bootstraps = [cls.get_bootstrap(name, ctx)
-                      for name in cls.list_bootstraps()]
-        acceptable_bootstraps = []
+                      for name in cls.all_bootstraps()]
+        acceptable_bootstraps = set()
+        known_web_packages = {"flask"}  # to pick webview over service_only
+        default_recipe_priorities = [
+            "webview", "sdl2", "service_only"  # last is highest
+        ]
```

Comment:
Just some thoughts, what if someone uses `flask` as a dep because they need a sub component of it, but still want to be `sdl2` or `service_only`. Probably in this case they can just force the boostrap passing it as param? Just curious/wondering

Reply:
SDL2 overrides webview in the special behavior, so that is no prob. The other way around is though, e.g. my website backend actually uses SDL2 :joy: but yeah in that case you just need to specify the bootstrap manually

Reply:
OK thanks for the info :+1: